### DISCUSSION TITLE
Code coverage monitoring and management towards 70% for 0.1.0 release

### DISCUSSION THREAD

**Discussion by haipinglu** [pykale/pykale#105] [Testing] [answered]
## Monitoring of progress towards 70% coverage

The following are modules that need tests, currently less than 70% from a [recent test](https://github.com/pykale/pykale/runs/2388932983) with a current coverage of **19%**. Also see the [Codecov report](https://codecov.io/gh/pykale/pykale/list/master/kale). We will have members assigned to these modules to move forward.

Name                                Stmts   Miss  Cover

- [ ] kale/embed/gcn.py                      98     98     0%   $~~~~~~~~~~~$             @pz-white 
- [ ] kale/embed/gripnet.py                  81     81     0%   $~~~~~~~~~~~$             @pz-white 
- [x] kale/embed/image_cnn.py               167    167     96%  $~~~~~~~~~~~$             @XianyuanLiu  
- [ ] kale/embed/seq_nn.py                   36     36     0%   $~~~~~~~~~~~$             @pz-white 
- [x] kale/embed/video_i3d.py               189    189     90%   $~~~~~~~~~~~$             @XianyuanLiu  
- [x] kale/embed/video_res3d.py             125    125     97%   $~~~~~~~~~~~$             @XianyuanLiu  
- [ ] kale/loaddata/dataset_access.py        16      6    62%
- [ ] kale/loaddata/multi_domain.py         111     65    41%
- [ ] kale/loaddata/sampler.py              161    131    20%
- [ ] kale/loaddata/tdc_datasets.py          28     28     0%   $~~~~~~~~~~~$             @pz-white
- [x] kale/loaddata/video_access.py          77     77     97%   $~~~~~~~~~~~$             @XianyuanLiu  
- [x] kale/loaddata/video_datasets.py        60     60     86%   $~~~~~~~~~~~$             @XianyuanLiu  
- [ ] kale/loaddata/videos.py                89     89     47%   $~~~~~~~~~~~$             @XianyuanLiu  
- [ ] kale/pipeline/deep_dti.py              49     49     0%   $~~~~~~~~~~~$             @pz-white
- [ ] kale/pipeline/domain_adapter.py       517    517     0%   $~~~~~~~~~~~$             @XianyuanLiu  
- [x] kale/predict/class_domain_nets.py      64     64     89%   $~~~~~~~~~~~$             @XianyuanLiu  
- [x] kale/predict/decode.py                 20     20     100%   $~~~~~~~~~~~$             @pz-white 
- [ ] kale/predict/isonet.py                204    204     0%   $~~~~~~~~~~~$             @haipinglu 
- [ ] kale/predict/losses.py                 52     52     0%
- [ ] kale/prepdata/chem_transform.py        31     31     0%   $~~~~~~~~~~~$             @pz-white 
- [x] kale/prepdata/prep_cmr.py              85     85     0%
- [x] kale/prepdata/video_transform.py       23     23     60%   $~~~~~~~~~~~$             @XianyuanLiu  
- [ ] kale/utils/csv_logger.py              169    169     0%                      $~~~~~~~~~~~$             @bobturneruk

**Comment by bobturneruk** [comment]
Each of these could be a separate issue and assigned accordingly.

**Reply by haipinglu** [reply]
> Each of these could be a separate issue and assigned accordingly.

Thanks for the suggestion! However, how many issues will be needed and who's going to do that? Bots? Are they really one of those four common issues? I'd rather spend the time to actually write tests.

The amount of "paper work" and (unnecessary) email traffics generated create quite some overhead for this scale of development that I found distracting and not helpful.

I think issues are more helpful for _external_ users to communicate with us. 

Nevertheless, I appreciate your suggestion.

**Comment by haipinglu** [comment, marked as answer]
Close this discussion since it is 73% now.
